crates/biome_service/src/file_handlers/graphql.rs

```diff
@@ -112,10 +125,313 @@ fn debug_syntax_tree(_rome_path: &BiomePath, parse: AnyParse) -> GetSyntaxTreeRe
+                    .into_iter()
+                    .map(biome_diagnostics::serde::Diagnostic::new)
+                    .collect::<Vec<_>>(),
+            );
+            let skipped_diagnostics = diagnostic_count.saturating_sub(diagnostics.len() as u32);
+
+            LintResults {
+                diagnostics,
+                errors,
+                skipped_diagnostics,
+            }
+        },
+    )
+}
+
+#[tracing::instrument(level = "debug", skip(params))]
+pub(crate) fn code_actions(params: CodeActionsParams) -> PullActionsResult {
+    let CodeActionsParams {
+        parse,
+        range,
+        workspace,
+        path,
+        manifest: _,
+        language,
+        settings,
+    } = params;
+    debug_span!("Code actions GraphQL", range =? range, path =? path).in_scope(move || {
+        let tree = parse.tree();
+        trace_span!("Parsed file", tree =? tree).in_scope(move || {
+            let rules = settings.as_rules(params.path.as_path());
+            let filter = rules
+                .as_ref()
+                .map(|rules| rules.as_enabled_rules())
+                .unwrap_or_default()
+                .into_iter()
+                .collect::<Vec<_>>();
+
+            let mut filter = AnalysisFilter::from_enabled_rules(filter.as_slice());
+
+            let mut categories = RuleCategoriesBuilder::default().with_syntax().with_lint();
+            if settings.organize_imports.enabled {
+                categories = categories.with_action();
+            }
+            filter.categories = categories.build();
+            filter.range = Some(range);
+
+            let analyzer_options = workspace.analyzer_options::<GraphqlLanguage>(path, &language);
+
+            let Some(_) = language.to_graphql_file_source() else {
+                error!("Could not determine the file source of the file");
+                return PullActionsResult { actions: vec![] };
+            };
+
+            trace!("GraphQL runs the analyzer");
+            let mut actions = Vec::new();
+
+            analyze(&tree, filter, &analyzer_options, |signal| {
+                actions.extend(signal.actions().into_code_action_iter().map(|item| {
+                    CodeAction {
+                        category: item.category.clone(),
+                        rule_name: item
+                            .rule_name
+                            .map(|(group, name)| (Cow::Borrowed(group), Cow::Borrowed(name))),
+                        suggestion: item.suggestion,
+                    }
+                }));
+
+                ControlFlow::<Never>::Continue(())
+            });
+
+            PullActionsResult { actions }
+        })
+    })
+}
+
+/// If applies all the safe fixes to the given syntax tree.
+pub(crate) fn fix_all(params: FixAllParams) -> Result<FixFileResult, WorkspaceError> {
+    let FixAllParams {
+        parse,
+        fix_file_mode,
+        workspace,
+        should_format: _, // we don't have a formatter yet
```

Comment:
You need to hurry up with this PR, or Denis would [like a word](https://github.com/biomejs/biome/pull/3296) :rofl: